design/open/high-level-python-wrapper.md

```diff
@@ -0,0 +1,314 @@
+
+
+class WriteIOBase:
+    def upload(self, data: bytes | str | TextIO | BinaryIO, path: str) -> ObjectInfo: ...
+    def import_from(self, object_store_uri: str, destination: str, commit_message: str) -> ImportManager: ...
+    def delete_objects(self, object_paths: str | Iterable[str]): ...
+    def copy_object(self, from_reference: str, from_path: str): ...
+    def open(self, path, mode: Literal['r', 'rb', 'w', 'wb'], pre_signed: Optional[bool] = None) -> TextIO | BinaryIO: ...
+
+
+class Branch(Reference, WriteIOBase):
+    def create(self, source_reference_id: str) -> Branch: ...
+    def head(self) -> Reference: ...
+    def commit_changes(self, message: str, metadata: dict[str, str]) -> Reference: ...
+    def reset(self, path: Optional[str] = None): ...
+    def reverse(self, reference_id: str): ...
+    def list_uncommitted(self, max_amount: Optional[int], after: str ='', prefix: str ='') -> Generator[Change]: ...
+    def delete(self): ...
+    def transaction(self, commit_message: str) -> BranchTransaction: ...
+    # override
+    def open(self, path, mode: Literal['r', 'rb', 'w', 'wb'], pre_signed: Optional[bool] = None) -> TextIO | BinaryIO: ...
+
+
+class Tag(Reference):
+    def delete(self): ...
+
+
+class CommonPrefix:
+    def __init__(self, repository_id: str, reference_id: str, path: str, client: Client = DefaultClient): ...
+    def path(self) -> self: ...
+    def exists(self) -> bool: ...
+
+
+class ObjectInfo:
+    def __init__(self, repository_id: str, reference_id: str, path: str, client: Client = DefaultClient): ...
+    def path(self) -> self: ...
+    def modified_time(self) -> datetime.datetime: ...
+    def size_bytes(self) -> int: ...
+    def content_type(self) -> Optional[str]: ...
+    def metadata(self) -> dict[str, str]: ...
+    def physical_address(self) -> str: ...
+    def delete(self): ...
+
+
+class Change(NamedTuple):
+    type: Literal(['added', 'removed', 'changed', 'conflict,', 'prefix_changed'])
+    path: str
+    path_type: Literal(['common_prefix', 'object'])
+    size_bytes: Optional[int]
+
+
+class ServerConfiguration:
+    def __init__(self, client: Client = DefaultClient): ...
+    def version(self) -> str: ...
+    def storage_config(self) -> ServerStorageConfiguration: ...
+
+
+class ServerStorageConfiguration(NamedTuple):
+    blockstore_type: str
+    pre_sign_support: bool
+    import_support: bool
+
+
+class ImportManager:
+    def __init__(self, repository_id: str, reference_id: str, client: Client = DefaultClient): ...
+    def start(self) -> str:
+        'start import, reporting back (and storing) a process id'
+        ...
+    def wait(self, poll_interval: timedelta = timedelta(seconds=2)) -> ImportResult:
+        'poll a started import task ID, blocking until completion'
+        ...
+    def run(self, poll_interval: Optional[timedelta] = None) -> ImportResult:
+        'same as calling start() and then wait()'
```

Comment:
Suggest we add method to describe the import's from as we support array of prefix:

```python
branch.import(commit_info)
  .prefix('s3://some/path', dest='optional/location')
  .object('s3://my/file')
  .run()

Reply:
Nice! adapted